src/gpu/generic/sycl/ref_convolution.hpp

```diff
@@ -0,0 +1,383 @@
+            }
+            return attr()->post_ops_.len() <= sycl_post_ops_t::max_post_ops
+                    && attr()->post_ops_.has_default_values(
+                            {primitive_kind::eltwise, primitive_kind::prelu,
+                                    primitive_kind::sum});
+        }
+
+        static bool check_data_types(const memory_desc_wrapper &src0,
+                const memory_desc_wrapper &src1,
+                const memory_desc_wrapper &dst) {
+            using namespace data_type;
+
+            const auto src0_dt = src0.data_type();
+            const auto src1_dt = src1.data_type();
+            const auto dst_dt = dst.data_type();
+
+            for (auto t : {src0_dt, src1_dt, dst_dt}) {
+                if (!utils::one_of(t, f32, bf16, f16, s32, s8, u8))
+                    return false;
+            }
+
+            return true;
+        }
+
+        static bool check_formats(const memory_desc_wrapper &src0,
+                const memory_desc_wrapper &src1,
+                const memory_desc_wrapper &dst) {
+            using namespace format_tag;
+
+            for (const auto &mdw : {src0, src1, dst}) {
+                if (!mdw.is_plain()) { return false; }
+            }
+            return true;
+        }
+
+        bool check_work_amount(const memory_desc_wrapper &weights) {
+            auto elems = weights.nelems();
+            auto work_per_output = elems / OC();
+            // arbitrarily chosen threshold to avoid unreasonably long runtimes
+            // such cases should use a different implementation
+            return work_per_output < 200000;
+        }
+    };
+
+    status_t init(impl::engine_t *engine) override;
+    status_t execute(const exec_ctx_t &ctx) const override;
+
+private:
+    const pd_t *pd() const { return (const pd_t *)primitive_t::pd().get(); }
+    kernel_t kernel_;
+};
+
+struct ref_convolution_bwd_weights_t : public gpu::generic::sycl::primitive_t {
+    using gpu::generic::sycl::primitive_t::primitive_t;
+
+    struct pd_t : public convolution_bwd_weights_pd_t {
+        using convolution_bwd_weights_pd_t::convolution_bwd_weights_pd_t;
+
+        DECLARE_COMMON_PD_T("dpcpp:ref:any", ref_convolution_bwd_weights_t);
+
+        status_t init(impl::engine_t *engine) {
+            using namespace data_type;
+            using sm = primitive_attr_t::skip_mask_t;
+
+            const memory_desc_wrapper data_d(src_md());
+            const memory_desc_wrapper diff_weights_d(diff_weights_md());
+            const memory_desc_wrapper diff_dst_d(diff_dst_md());
+
+            const bool ok = is_bwd_w() && check_work_amount(diff_weights_d)
+                    && set_default_formats()
+                    && check_data_types(data_d, diff_weights_d, diff_dst_d)
+                    && check_formats(data_d, diff_weights_d, diff_dst_d)
+                    && attr()->has_default_values(sm::scales_runtime
+                            | sm::zero_points_runtime | sm::post_ops
```

Comment:
post ops are reserved for forward computation, since there is no gradient computation semantics for post-ops in oneDNN.